File: src/main/java/gestion/service/EnseignantService.java
package gestion.service;

import java.util.List;

import gestion.dto.EnseignantDTO;

public interface EnseignantService {

	public List<EnseignantDTO> getList();
	
	public EnseignantDTO load(Long id);
	
	public EnseignantDTO save(EnseignantDTO dto);
	
	public void delete(Long id);
	
	public Long getCount();

}

File: web/src/api/EnseignantService.js
import axios from 'axios';

const Enseignant_API_BASE_URL = "http://localhost:8080/gestion/enseignant/rest";

class EnseignantService
{

    getEnseignants(id)
    {
        return axios.get(Enseignant_API_BASE_URL + '/list', { groupId: id });
    }

    createEnseignant(Enseignant)
    {
        return axios.post(Enseignant_API_BASE_URL + '/save/', Enseignant);
    }

    getEnseignantById(EnseignantId)
    {
        return axios.get(Enseignant_API_BASE_URL + '/load/' + EnseignantId);
    }

    updateEnseignant(Enseignant)
    {
        return axios.put(Enseignant_API_BASE_URL + '/save', Enseignant);
    }

    deleteEnseignant(EnseignantId)
    {
        return axios.delete(Enseignant_API_BASE_URL + '/delete/' + EnseignantId);
    }
}

export default new EnseignantService()